File: Arrays/SquaresofaSortedArray.py
class Solution(object):
    def sortedSquares(self, nums):
        for i in range(len(nums)):
            if(nums[i]<0):
                nums[i]=nums[i]*(-1)
        nums.sort()
        for i in range(len(nums)):
            nums[i]=nums[i]*nums[i]
        return nums
        
                
        """
        :type nums: List[int]
        :rtype: List[int]
        """

File: squares-of-a-sorted-array/squares-of-a-sorted-array.cpp
class Solution {
public:
    vector<int> sortedSquares(vector<int>& nums) {
        int r,l,k;
        int n= nums.size();
        vector <int> ans(n, 0);
        l=0;
        r=n-1;
        k=n-1;
        while(l<=r)
        {
            if(abs(nums[l])<abs(nums[r]))
            {
                ans[k--]=nums[r]*nums[r];
                r--;
            }
            else
            {
                ans[k--]=nums[l]*nums[l];
                l++;
            }
            
        }
        
        return ans;
    }
};


